BraintreeCore/src/main/java/com/braintreepayments/api/core/ErrorWithResponse.kt

```diff
@@ -98,18 +105,19 @@ data class ErrorWithResponse internal constructor(
             try {
                 val errors = json
                     ?.let { JSONObject(it) }
-                    ?.getJSONArray(GraphQLConstants.Keys.ERRORS)
+                    ?.optJSONArray(GraphQLConstants.Keys.ERRORS)
+                    ?: throw JSONException("Errors key not found in JSON")
```

Comment:
We don't typically throw `JSONException` in the SDK. Is there a big advantage of using `optJSONArray` instead of `getJSONArray` as before?

Reply:
-> Returns null: Instead of throwing an exception, optJSONArray() will return null if the key is not found or the value is not a valid JSONArray.
-> Graceful handling: It provides a more flexible way to access optional or uncertain data. This can be useful when you are dealing with data that may not always include the key or if the key's value may not be a JSONArray but you don't want to handle exceptions.
->Use optJSONArray() if we prefer more flexible code, where the absence of the key or the wrong type doesn't cause an error but can be handled in a more graceful way

Reply:
> If we don't typically throw JSONException in the SDK, then is okay to return null or empty string?

This might be tricky. I'd defer to whatever iOS does in that case. iOS doesn't use exceptions but I believe they do have nullability checks for either of the fields to determine how the error response is parsed.

Overall this comment is a take-or-leave nitpick. The only reason i'm calling it out is it does feel somewhat odd to throw an exception type we don't own.

Reply:
+1 
I agree, to handling as iOS is doing!!